service/StudentService.java

```diff
@@ -0,0 +1,51 @@
+package service;
+
+import dao.Dao;
+import model.Student;
+
+import java.util.Iterator;
+
+public class StudentService {
+    private final Dao<Student> studentDao;
+
+    public StudentService(Dao<Student> studentDao){
+        this.studentDao = studentDao;
+    }
+
+    public Student getStudentById(String id){
+        Student student = studentDao.get(id).get();
+        return student;
+    }
+
+    public void getAllStudents(){
+        System.out.println();
+        Iterator<Student> iterator = studentDao.getAll().iterator();
+        System.out.printf("%-12s %-20s %s%n", "StudentId", "FirstName", "LastName");
+        while (iterator.hasNext()){
+            Student student = iterator.next();
+            System.out.printf("%-12s %-20s %s%n",student.getStudentId(),student.getFirstName(),student.getLastName() );
+        }
+    }
+
+    public void addStudent(Student student){
+        studentDao.save(student);
+    }
+
+    public void updateStudent(Student student,String[] params){
+        studentDao.update(student,params);
+    }
+
+    public void deleteStudent(Student student){
+        studentDao.delete(student);
+    }
+
+    public void showMenu(){
```

Comment:
This should be part of the `Presentation Layer`, not the `Service Layer`. You could move it to a separate class called `Menu` for example. But this `Menu` class then should be inside the `Presentation Layer`.